### PR TITLE
Update secret api

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-secrets/src/secret_serializers.cpp [hunk 1]
```diff
@@ -152,3 +152,38 @@ void KeyVaultDeletedSecretSerializer::KeyVaultDeletedSecretDeserialize(
   secret.DeletedDate
       = PosixTimeConverter::PosixTimeToDateTime(jsonParser[_detail::DeletedDatePropertyName]);
 }
+
+std::string KeyVaultSecretPropertiesSerializer::KeyVaultSecretPropertiesSerialize(
+    KeyvaultSecretProperties const& properties)
+{
+  Azure::Core::Json::_internal::json payload;
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/src/secret_serializers.cpp:159]: nit. 
add `using namespace Azure::Core::Json::_internal` to the top of the file so this line and all other can be shortened as `json payload` only

**Review by vhvb1989 — COMMENTED** [review]
some comments

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 2]
```diff
@@ -155,6 +155,44 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
     Azure::Response<KeyVaultDeletedSecret> GetDeletedSecret(
         std::string const& name,
         Azure::Core::Context const& context = Azure::Core::Context()) const;
+
+    /**
+     * @brief Updates the attributes associated with a specified secret in a given key vault.
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:188]: done, cool thing is , this is copy paste from swagger

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 3]
```diff
@@ -183,6 +195,46 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
...
+     * @param name The name of the secret<span class="x x-first x-last">.</span>
+     * @param options The optional parameters for this request.
+     * @param properties The properties to update
+     * @param context The context for the operation can be used for request cancellation.
+     *
+     * @return The Secret wrapped in the Response.
+     */
+    Azure::Response<KeyVaultSecret> UpdateSecretProperties(
+        std::string const& name,
+        UpdateSecretPropertiesOptions const& options,
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:215]: usually the clientOptions model is an optional param like the Context.  You might want to check how other languages are handling this for `UpdateSecret...`  API.

**Review by vhvb1989 — APPROVED** [review]
I think is fine for now and to keep adding more content.
But, we might need to come back and review the UpdatePropertiesOptions argument in the future

**Pull request merged**